static/app/views/performance/database/noDataMessage.tsx

```diff
@@ -1,106 +1,120 @@
-
-  if (areSpanMetricCountsLoading || areIneligibleProjectsFetching) {
+  if (isDataFetching) {
     return null;
   }
 
-  if (!projectSpanMetricsCounts) {
-    return null;
-  }
+  const hasAnyProblematicProjects =
+    (!areOutdatedProjectsFetching && outdatedProjects.length > 0) ||
+    (!areDenylistProjectsFetching && denylistedProjects.length > 0);
 
-  if (doesAnySelectedProjectHaveMetrics) {
+  if (isDataAvailable && !hasAnyProblematicProjects) {
     return null;
   }
 
-  const firstIneligibleProjects = ineligibleProjects.slice(0, MAX_LISTED_PROJECTS + 1);
-
   return (
     <Wrapper>
-      {t('No queries found.')}{' '}
-      {tct(
-        'Try updating your filters, or learn more about performance monitoring for queries in our [documentation:documentation].',
-        {
-          documentation: (
-            <ExternalLink href="https://docs.sentry.io/product/performance/queries/" />
-          ),
-        }
-      )}{' '}
-      {ineligibleProjects.length > 0 &&
-        tct('You may also be missing data due to outdated SDKs: [projectList]', {
-          documentation: (
-            <ExternalLink href="https://docs.sentry.io/product/performance/query-insights/" />
-          ),
-          projectList: (
-            <Fragment>
-              {firstIneligibleProjects.map((project, projectIndex) => {
-                return (
-                  <span key={project.id}>
-                    <a
-                      href={normalizeUrl(
-                        `/organizations/${organization.slug}/projects/${project.slug}/`
-                      )}
-                    >
-                      {project.name}
-                    </a>
-                    {projectIndex < firstIneligibleProjects.length - 1 && ', '}
-                  </span>
-                );
-              })}
-            </Fragment>
-          ),
-        })}
-      {hasMoreIneligibleProjectsThanVisible &&
-        tct(' and [count] more.', {
-          count: ineligibleProjects.length - MAX_LISTED_PROJECTS,
+      {!isDataAvailable &&
+        tct(
+          'No queries found. Try updating your filters, or learn more about performance monitoring for queries in our [documentation:documentation].',
+          {
+            documentation: (
+              <ExternalLink href="https://docs.sentry.io/product/performance/queries/" />
+            ),
+          }
+        )}{' '}
+      {outdatedProjects.length > 0 &&
+        tct('You may be missing data due to outdated SDKs: [projectList].', {
+          projectList: <ProjectList projects={outdatedProjects} />,
         })}{' '}
+      {denylistedProjects.length > 0 &&
+        tct(
+          'Some of your projects have been omitted from query performance analysis. Please [supportLink]. Omitted projects: [projectList].',
+          {
+            supportLink: (
+              <Button priority="link" onClick={() => openHelpSearchModal({organization})}>
+                {t('contact support')}
```

Comment:
Not sure between `Contact support` and `Contact Support`, just took my best guess at style.

```suggestion
                {t('Contact Support')}
```